app/Http/Controllers/App/PostController.php

```diff
@@ -49,11 +51,28 @@ public function store(Request $request)
             'team_id' => ['required', 'exists:teams,id', 'in:'.$request->user()->teams()->whereNotNull('approved_at')->pluck('id')->join(',')],
             'banner_id' => ['nullable', 'exists:banners,id'],
             'challenge_id' => ['nullable', 'exists:challenges,id'],
+            'image' => ['nullable', 'file', 'image'],
         ]);
 
         $author = $request->user();
         $data['author_id'] = $author->id;
 
+        if (!empty($data['image'])) {
+            // save original image
+            $imageOriginal = Image::make($data['image']);
+            $filename = 'post/originals/' . $data['image']->hashName();
+            Storage::disk('upload')->put($filename, $imageOriginal->encode());
+
+            // save preview version
+            $imagePreview = Image::make($data['image'])->resize(1024, 1024, function ($constraint) {
+                $constraint->aspectRatio();
```

Comment:
I think we should prevent upscaling as well (don't resize the image, if it is small than 1024x1024):
```suggestion
                $constraint->aspectRatio();
                $constraint->upsize();
```